pkg/manager/member/pd_scaler.go

```diff
@@ -39,27 +39,36 @@ func NewPDScaler(pdControl pdapi.PDControlInterface,
 	return &pdScaler{generalScaler{pdControl, pvcLister, pvcControl}}
 }
 
+func (psd *pdScaler) Scale(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	if *newSet.Spec.Replicas > *oldSet.Spec.Replicas {
+		return psd.ScaleOut(tc, oldSet, newSet)
+	} else if *newSet.Spec.Replicas < *oldSet.Spec.Replicas {
+		return psd.ScaleIn(tc, oldSet, newSet)
+	}
+	return nil
+}
+
 func (psd *pdScaler) ScaleOut(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	_, ordinal, replicas, deleteSlots := scaleOne(oldSet, newSet)
```

Comment:
recalculating again because of less code needs to be changed (otherwise ScaleOut/ScaleIn function signatures need to change and a lot of tests must be updated)